src/python/pants/engine/internals/native.py

```diff
@@ -205,32 +230,13 @@ def new_scheduler(
             ignore_patterns,
             use_gitignore,
             root_subject_types,
-            # Remote execution config.
-            execution_options.remote_execution,
-            execution_options.remote_store_server,
-            execution_options.remote_execution_server,
-            execution_options.remote_execution_process_cache_namespace,
-            execution_options.remote_instance_name,
-            execution_options.remote_ca_certs_path,
-            execution_options.remote_oauth_bearer_token_path,
-            execution_options.remote_store_thread_count,
-            execution_options.remote_store_chunk_bytes,
-            execution_options.remote_store_connection_limit,
-            execution_options.remote_store_chunk_upload_timeout_seconds,
-            execution_options.remote_store_rpc_retries,
-            tuple(
-                tuple(pair.split("=", 1))
-                for pair in execution_options.remote_execution_extra_platform_properties
-            ),
+            remoting_options,
             execution_options.process_execution_local_parallelism,
             execution_options.process_execution_remote_parallelism,
```

Comment:
I thought about going further with this PR than it currently does. Although in that case, my thought was that the process execution options should be a second struct because they differ in focus from `RemotingOptions`. Specifically, the process execution config focuses on execution strategy between local and remote, while `RemotingOptions` focuses on configuring remote execution.

Moreover, `ExecutionOptions` on the Python side would go away and the Python code would just directly instantiate the `PyRemotingOptions` and `PyExecutionStrategyOptions`. This would be beneficial because it would reduce duplicate data definitions but also allow putting the defaults into Rust code closer to the places they are used. (Then the Python<->Rust layer exposes `PyRemotingOptions::default` as a `default` class method used by the options system.

Reply:
> ExecutionOptions on the Python side would go away 

That might be challenging, unfortunately, due to import cycles between the native code and options parsing. We use some portions of the options code to even build the native code via https://github.com/pantsbuild/pants/blob/master/build-support/bin/native/cargo.sh#L31 . It would be great not to... it might just require duplicating the logic that `download_binary.sh` is trying to reuse.

Reply:
In the meantime, what are your thoughts on this PR and the introduction in a subsequent PR of a similar `ExecutionStrategyOptions` struct? There is an incremental benefit from just this change.